src/targets/php/http2/client.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @description
+ * HTTP code snippet generator for PHP using curl-ext.
+ *
+ * @author
+ * @AhmadNassri
+ *
+ * for any questions or issues regarding the generated code snippet, please open an issue mentioning the author.
+ */
+
+import { CodeBuilder } from '../../../helpers/code-builder';
+import { getHeader, getHeaderName, hasHeader } from '../../../helpers/headers';
+import { Client } from '../../targets';
+import { convertType } from '../helpers';
+
+export interface Http2Options {
+  closingTag?: boolean;
+  noTags?: boolean;
+  shortTags?: boolean;
+}
+
+export const http2: Client<Http2Options> = {
+  info: {
+    key: 'http2',
+    title: 'HTTP v2',
+    link: 'http://devel-m6w6.rhcloud.com/mdref/http',
+    description: 'PHP with pecl/http v2',
+  },
+  convert: ({ postData, headersObj, method, queryObj, cookiesObj, url }, options) => {
+    const opts = {
+      closingTag: false,
+      indent: '  ',
+      noTags: false,
+      shortTags: false,
+      ...options,
+    };
+
+    const { push, blank, join } = new CodeBuilder({ indent: opts.indent });
+    let hasBody = false;
+
+    if (!opts.noTags) {
+      push(opts.shortTags ? '<?' : '<?php');
+      blank();
+    }
+
+    push('$client = new http\\Client;');
+    push('$request = new http\\Client\\Request;');
+    blank();
+
+    switch (postData.mimeType) {
+      case 'application/x-www-form-urlencoded':
+        push('$body = new http\\Message\\Body;');
+        push(
+          `$body->append(new http\\QueryString(${convertType(postData.paramsObj, opts.indent)}));`,
+        );
+        blank();
+        hasBody = true;
+        break;
+
+      case 'multipart/form-data': {
+        const files: {
+          name: string;
+          type: string | undefined;
+          file: string;
+          data: string;
+          [anything: string]: string | undefined;
+        }[] = [];
+        const fields: Record<string, any> = {};
+        postData.params?.forEach(({ name, fileName, value, contentType }) => {
+          if (fileName) {
+            files.push({
+              name,
+              type: contentType,
+              file: fileName,
+              // @ts-expect-error appears to be a genuine error
```

Comment:
What's the error here?

Reply:
good catch: I think when I ignored this it was at a later moment and I assumed this error came from elsewhere when it came right from like 2 lines above all along!  fixed in 6f26447bbf0edddc05f5d89c890b134f4bb37b6a